### ISSUE TITLE
Apple Pay "We were not able to verify your domain" admin notice showing up

### ISSUE THREAD

**Issue by a-danae** [Automattic/woocommerce-payments#4712]
### Describe the bug

An Apple Pay warning admin notice is displayed on the admin, regardless of whether the steps provided by the notice are actually needed. This is happening since 4.7.0.

The admin notice on the WCPay settings page reads "Apple Pay: We were not able to verify your domain. [This help documentation](https://woocommerce.com/document/payments/apple-pay/#apple-pay-button-does-not-appear) will walk you through the process to verify with Apple that you control your domain". 

Payments made with Apple Pay should continue to work as long as the configuration file Apple Pay looks for exists, regardless of the presence of the admin notice. More details about this are below in the "Expected behavior" section.

### To Reproduce

1. Go to Payments -> Settings
2. Under Express Checkouts, enable "Apple Pay / Google Pay", if not enabled already. Save and reload the page
3. Notice the notice at the top of the page

### Actual behavior

Apple Pay's warning shows up when it's not needed. That is when either the configuration file exists or the option under Admin > Settings > Permalinks has anything but "Plain" selected.

### Screenshots

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/41606954/188957897-2cc22140-9d25-4f98-a9b6-00aa49ee6fe4.png">

### Expected behavior
We use custom rewrites to generate the file Apple Pay requires when this option in Admin > Settings > Permalinks is anything but "Plain". The notice shouldn't show up in this case.
When the option under Admin > Settings > Permalinks has "Plain" selected and no configuration file exists, the notice should show up. 

### Additional context
This issue is high priority because following the steps provided by the warning could lead merchants to mess up their Apple Pay flow.

Introduced in https://github.com/Automattic/woocommerce-payments/pull/4551

**Comment by gabriel-fuentes** [comment]
This is also happening for a WordPress.com user: `36693213-hc`

**Comment by roryWolf** [comment]
Adding in the following merchant who is also experiencing the issue:  5541003-zen

**Comment by csmcneill** [comment]
5549035-zen

Details:
- WPCOM site
- Apple Pay only shows up when `Allow websites to check for Apple Pay and Apple Card` is disabled under **Safari > Preferences > Privacy** (related issue: #3876)
- Permalink settings are not `Plain`.
- The `.well-known/apple-developer-merchantid-domain-association` is available at the proper URL.
- When `Allow websites to check for Apple Pay and Apple Card` is disabled, the following notice shows up in the console:

```
Either you do not have a card saved to your Wallet or the current domain ([site]) or stripeAccount parameter (acct_xxx) is not registered for Apple Pay. Visit https://dashboard.stripe.com/account/apple_pay to register this domain.
```

- When `Allow websites to check for Apple Pay and Apple Card` is enabled, the following notice shows up in the console:

```
Could not create Apple Pay session. Please make sure you have registered this Stripe account. For more information, see https://stripe.com/docs/apple-pay#web.
```

**Comment by a-danae** [comment]
This issue was solved by https://github.com/Automattic/woocommerce-payments/pull/4758, and released in WCPay 4.7.1.
